src/polling.ts

```diff
@@ -90,41 +90,60 @@ export class PollingProcessor {
       });
   }
 
-  private async retrieveFlags(): Promise<void> {
-    try {
-      this.log.debug('Fetching flags started');
-      const response = await this.api.getFeatureConfig(
-        this.environment,
-        this.cluster,
-      );
-      this.log.debug('Fetching flags finished');
-      response.data.forEach((fc: FeatureConfig) =>
-        this.repository.setFlag(fc.feature, fc),
-      );
-    } catch (error) {
-      this.log.error('Error loading flags', error);
-      throw error;
+  private async retrieveFlags(retries = 3): Promise<void> {
+    for (let i = 0; i < retries; i++) {
+      try {
+        this.log.debug('Fetching flags');
+        const response = await this.api.getFeatureConfig(
+          this.environment,
+          this.cluster,
+        );
+        this.log.debug('Fetching flags finished');
+        response.data.forEach((fc: FeatureConfig) =>
+          this.repository.setFlag(fc.feature, fc),
+        );
+        break;
```

Comment:
Not a problem, but returning from the function might be a better option than breaking the loop. The break could, ummm, break, if another breakable scope wraps this, where as the return is pretty definitive.

Reply:
Updated to return in this function and it's segment counterpart

Reply:
https://github.com/harness/ff-nodejs-server-sdk/commit/b72eb5040ff7c227d0a7250e1c7df924581ca490